src/toil/batchSystems/mesos/executor.py

```diff
@@ -102,12 +147,33 @@ def launchTask(self, driver, task):
         def runTask():
             log.debug("Running task %s", task.task_id.value)
             sendUpdate(mesos_pb2.TASK_RUNNING)
-            # This is where task.data is first invoked. Using this position to setup cleanupInfo
             taskData = pickle.loads(task.data)
             if self.workerCleanupInfo is not None:
                 assert self.workerCleanupInfo == taskData.workerCleanupInfo
+                assert self.masterAddress == taskData.masterAddress
+                assert self.mesosCredentials == taskData.mesosCredentials
             else:
+                # This is where task.data is invoked for the firs ttiem on this executor. Using this
+                # position to setup workerCleanupInfo, masterAddress, mesosCredentials, and
+                # deadSpace for this executor.
                 self.workerCleanupInfo = taskData.workerCleanupInfo
+                self.masterAddress = taskData.masterAddress
```

Comment:
So getting back to this, I installed Mesos on my laptop and tried to use pycharm debugging to pause the execution of the code and poke around the attributes of driver, executorInfo, frameworkInfo, and slaveInfo -- variables passed to `registered`. driver doesn't have anything (pycharm just shows it as a class object with no variables) and the others don't have anything that looks like the resolved master address.